src/Agent.Worker/Build/TrackingConfig.cs

```diff
@@ -65,20 +64,31 @@ public TrackingConfig(IExecutionContext executionContext, ServiceEndpoint endpoi
         [JsonProperty("agent_builddirectory")]
         public string BuildDirectory { get; set; }
 
-        public string CollectionName { get; set; }
+        public string CollectionUrl { get; set; }
 
         public string DefinitionName { get; set; }
 
         [JsonProperty(FileFormatVersionJsonProperty)]
         public int FileFormatVersion
         {
-            get { return 2; }
+            get
+            {
+                return 3;
+            }
 
             set
             {
-                if (value != 2)
+                // Version 3 changes:
```

Comment:
There is very little change between the version 2 and 3 file format version. Just the CollectionName/Url. The file gets upgraded to the new version format every time it is rewritten - i.e. every build or when marked for GC.